src/foam/dao/PipelinePMDAO.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao;
+
+import foam.core.FObject;
+import foam.core.X;
+import foam.mlang.order.Comparator;
+import foam.mlang.predicate.Predicate;
+import foam.nanos.pm.PM;
+
+public class PipelinePMDAO
+  extends ProxyDAO
+{
+  protected String putName_;
+  protected String findName_;
+  protected String removeName_;
+  protected String removeAllName_;
+
+  public PipelinePMDAO(X x, DAO delegate) {
```

Comment:
Also needs the daoKey or name of the DAO to report. Otherwise something like AuthenticateDAO will have the same PM key for all DAO decorator chains and you won't know which one you're recording, or rather, you'll actually be combining the results from different DAO's together.

Reply:
I think the init method makes this distinction when creating the names by appending the name of the delegate before the type of operation. I will however refactor this into its own variable to make things cleaner and more explicit.